src/AI/agents/agents/services/llm/intent_parser.py

```
@@ -0,0 +1,164 @@
+
+def validate_goal_safety(goal: str) -> tuple[bool, Optional[str]]:
+    """Comprehensive safety check for user goals"""
+    
+    # First, quick safety check without LLM
+    is_safe, safety_reason = _validate_goal_safety_quick(goal)
+    if not is_safe:
+        return False, safety_reason
+    
+    try:
+        # Full LLM-based parsing and validation
+        parsed = parse_intent(goal, attachments=attachments)
```

Ruff (0.14.8)

Undefined name `attachments`

(F821)